pkg/opensearch/client/client.go

```diff
@@ -126,9 +127,11 @@ var NewClient = func(ctx context.Context, ds *backend.DataSourceInstanceSettings
 
 	version, err := extractVersion(jsonData.Get("version"))
 	if err != nil {
-		return nil, fmt.Errorf("opensearch version is required, err=%v", err)
+		return nil, fmt.Errorf("version is required, err=%v", err)
 	}
 
+	flavor := jsonData.Get("flavor").MustString(string(OpenSearch))
+
 	timeField, err := jsonData.Get("timeField").String()
 	if err != nil {
 		return nil, fmt.Errorf("opensearch time field name is required, err=%v", err)
```

Comment:
Should this also be:

```suggestion
		return nil, fmt.Errorf("time field name is required, err=%v", err)
```

Also at line 114: 

```diff
- return nil, fmt.Errorf("error reading opensearch version")
+ return nil, fmt.Errorf("error reading version")
```